cli/tools/registry/mod.rs

```diff
@@ -635,6 +639,46 @@ async fn publish_package(
     package.package,
     package.version
   );
+
+  if provenance {
+    // Get the version manifest from JSR
+    let meta_url = jsr_url().join(&format!(
+      "@{}/{}/{}_meta.json",
+      package.scope, package.package, package.version
+    ))?;
+
+    let meta_bytes = client.get(meta_url).send().await?.bytes().await?;
+
+    // TODO: Verify manifest.
+
+    let subject = provenance::Subject {
+      name: format!("{}/{}", package.scope, package.package),
+      digest: provenance::SubjectDigest {
+        sha512: hex::encode(sha2::Sha512::digest(&meta_bytes)),
```

Comment:
```suggestion
        sha512: hex::encode(sha2::Sha512::digest(&meta_bytes)),
```

Use `sha256`. Easy to compare with hash in lockfile.